File: projectmanager/frontend/src/app/services/task/task.ts
export class Task {

  readonly projectId;
  readonly parentId;
  readonly taskName: string;
  readonly startDate: string;
  readonly endDate: string;
  readonly priority;
  status;
  taskId: string

  constructor(projectId, parentId, taskName, startDate, endDate, priority) {
    this.projectId = projectId;
    this.parentId = parentId;
    this.taskName = taskName;
    this.startDate = startDate;
    this.endDate = endDate;
    this.priority = priority;
  }

  public setTaskId(taskId: string) {
    this.taskId = taskId;
  }

  public setStatus(status: string) {
    this.status = status;
  }

}

File: projectmanager/src/main/java/com/cognizant/assignments/casestudy/entity/Task.java
package com.cognizant.assignments.casestudy.entity;

import java.io.Serializable;
import java.time.LocalDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.Data;

@Data
@Entity
@Table(name = "TASK")
public class Task implements Serializable {

	private static final long serialVersionUID = -279394459515423554L;
	
	@Id
    @Column(name = "TASK_ID", nullable = false, updatable = false)
	private int taskId;
	
	@Column(name = "PARENT_ID", nullable = false)
    private int parentId;
	
	@Column(name = "PROJECT_ID", nullable = false)
    private int projectId;
	
	@Column(name = "TASK", nullable = false)
    private String task;
	
	@Column(name = "START_DATE", nullable = false)
	private LocalDate startDate;
	
	@Column(name = "END_DATE", nullable = false)
	private LocalDate endDate;
	
	@Column(name = "PRIORITY", nullable = false)
    private int priority;
	
	@Column(name = "STATUS", nullable = false)
    private String status;	

}
